### PR TITLE
Issue #20 - Handle RPC return codes appropriately

### PR DESCRIPTION
**Pull request by dmacattack**
- enum mapping function to provide granular conversion

### CODE REVIEW

File: up-transport-vsomeip/src/message_conversions.rs
Comment (PLeVasseur) [inline up-transport-vsomeip/src/message_conversions.rs:263]: Apologies @dmacattack -- I should have posted [this portion](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/pull/22) of the SOME/IP spec in task #20.

File: up-transport-vsomeip/src/message_conversions.rs
Comment (PLeVasseur) [inline up-transport-vsomeip/src/message_conversions.rs:248]: This handles UMessage commstatus -> vsomeip message status mapping.

We also need to handle vsomeip message status -> UMessage commstatus mapping over [here](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/blob/acbb0d0c9b8b48dd35c74f461e97151f1e922000/up-transport-vsomeip/src/message_conversions.rs#L507) as mentioned in the task following the same [spec](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/pull/22).

File: up-transport-vsomeip/src/message_conversions.rs
Comment (PLeVasseur) [inline up-transport-vsomeip/src/message_conversions.rs:237]: I like this and getting rid of the `if ok {}` block below! :+1:

**Review by PLeVasseur — CHANGES_REQUESTED** [review]
Hey @dmacattack, I missed linking the spec in the task, please see comments.

File: up-transport-vsomeip/src/message_conversions.rs
Comment (dmacattack) [reply up-transport-vsomeip/src/message_conversions.rs:248]: > We also need to handle vsomeip message status -> UMessage ...
my mistake. Brain skipped over

File: up-transport-vsomeip/src/message_conversions.rs
Comment (PLeVasseur) [inline up-transport-vsomeip/src/message_conversions.rs:603]: Looks like it matches spec :+1:

File: up-transport-vsomeip/src/message_conversions.rs
Comment (PLeVasseur) [inline up-transport-vsomeip/src/message_conversions.rs:266]: @stevenhartley -- so mapping over from UCode to SOME/IP Error Code for those items in [the table](https://github.com/eclipse-uprotocol/up-transport-zenoh-rust/pull/72) there are cases where there are two possible SOME/IP Error Codes.

Which one should be chosen when mapping UCode => SOME/IP Error Code?

Comment (PLeVasseur) [reply up-transport-vsomeip/src/message_conversions.rs:266]: This shouldn't hold up the PR, but it might be good if you add a // TODO here and then open an issue on the up-spec repo to clarify intention.

**Review by PLeVasseur — APPROVED** [review]
Looks good.

Unsure why CI didn't run. I'm going to approve to see if that'll run the CI to make sure all is good.

File: up-transport-vsomeip/src/message_conversions.rs
Comment (dmacattack) [reply up-transport-vsomeip/src/message_conversions.rs:266]: food for thought: If the enums were bitvectors, we could use "OR" and encapsulate all the edge cases.

Comment (PLeVasseur) [reply up-transport-vsomeip/src/message_conversions.rs]: Interesting thought, done some of these in my day. How would this work with one side being a Protobuf enum and the other being a C enum that we have to map back down into what vsomeip is expecting?

Comment (dmacattack) [reply up-transport-vsomeip/src/message_conversions.rs]: AFAIK protobuf enums support non-sequential, and reserved enumerated values. Effectively you have to be able to change both sides for it to work correctly. VSomeIP mapping is basically 1:1, UCode mapping can be 1:2. 

pseudo eg:
``` 
// vsomeip values are bit unique
enum vsomeip_e { 
   E_OK = 0,
   E_WRONG_MESSAGE_TYPE  = 0x01,
   E_UNKNOWN_METHOD = 0x02,
   // NOTE value 0x03 is not allowed
   E_TIMEOUT = 0x04
   ...
};

enum UCode {
   OK = E_OK, // 00
   WRONG_MSG_TYPE = E_WRONG_MESSAGE_TYPE, // 0x01
   UNKNOWN_METHOD=   E_UNKNOWN_METHOD,  // 0x02
   INVALID_ARGUMENT  = ( WRONG_MSG_TYPE | UNKNOWN_METHOD) // 0x03
};

```

Comment (PLeVasseur) [reply up-transport-vsomeip/src/message_conversions.rs]: I see where you're coming from. Let's leave as-is for now in this PR.

**Pull request merged**
